File: 242-valid-anagram/valid-anagram.rs
use std::collections::HashMap;

impl Solution {
    pub fn is_anagram(s: String, t: String) -> bool {
        if s.len() != t.len() {
            return false;
        }

        let mut s_count = HashMap::new();
        let mut t_count = HashMap::new();

        for c in s.chars() {
            *s_count.entry(c).or_insert(0) += 1;
        }

        for c in t.chars() {
            *t_count.entry(c).or_insert(0) += 1;
        }

        s_count == t_count
    }
}

File: 242-valid-anagram/valid-anagram.py
class Solution:
    def isAnagram(self, s: str, t: str) -> bool:
        tracker = [0] * 26

        for letter in s:
            tracker[ord(letter)-ord('a')] += 1

        for letter in t:
            tracker[ord(letter)-ord('a')] -= 1

        for letter in tracker:
            if letter != 0:
                return False

        return True